lib/clerk/authenticate_request.rb

```diff
@@ -61,10 +61,14 @@ def resolve_cookie_token(env)
 
       return resolve_handshake(env) if auth_context.handshake_token?
 
-      if auth_context.development_instance? && auth_context.dev_browser?
+      if auth_context.development_instance? && !!auth_context.dev_browser_in_url
```

Comment:
:question: Can you please explain this change? 

We're triggering a handshake for 

- development instance with dev browser in URL
- development instance without dev browser

What's the rest of the cases for development instances that won't trigger a handshake?

Reply:
In our latest Core 2 ClerkJS version, we are setting the devBrowser to a cookie. So in the happy path where the customer application has been loaded and a devBrowser exists the request state depends on the __session & __client_uat cookies state (not on the dev browser).

> development instance with dev browser in URL

The most common case for this is when sign-in or sign-up is completed in the AP and the user is redirected back to their application with a dev browser. In this case we need to trigger the handshake to sync the state of FAPI with the app.